pkg/operator/manager.go

```diff
@@ -46,7 +46,7 @@ func RunOperator(cmd *cobra.Command, args []string) {
 
 	// Create a new Cmd to provide shared dependencies and start components
 	mgr, err := manager.New(config, manager.Options{
-		Namespace:          options.Namespace,
+		Namespace:          options.WatchNamespace(),
```

Comment:
setting `Namespace=""` here will set the watch namespace for all resources and this is the reason other permissions are added to the cluster role for other resources (pods, services, etc.)
another option to solve that is to add a differnet manger that handles cluster wide resources.